File: MssPayProSaasFront/src/app/modules/user/user.dto.ts
// File Path: src/app/modules/user/user.dto.ts

import { GroupDto } from './../group/group.dto';

export class UserDTO {
  id?: any;
  userName?: string;
  username?: string;
  emailId?: string;
  email?: string;
  password?: string;
  firstname?: string;
  firstName?: string;
  lastName?: string;
  lastname?: string; // Added this line
  keycloakId?: any;
  groups?: GroupDto[];
  pictureUrl?: any;
  roles?: string[];
  superUser: boolean = false;
}

File: facture/demo/src/main/java/com/mss/demo/entity/dto/UserDTO.java
package com.mss.demo.entity.dto;

import lombok.Data;
import java.util.List;

@Data
public class UserDTO {
    private String id; // String type
    private String username;
    private String firstname;
    private String lastname;
    private String email;
    private Long groupId;
    private List<GroupDTO> groups;
    private String keycloakId; // Added keycloakId field
}
